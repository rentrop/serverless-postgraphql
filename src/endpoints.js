
var endpoint = 'http://localhost:5000/graphql';

if(process.env.USE_AWS_ENDPOINT) {
  endpoint = process.env.AWS_ENDPOINT;
};

module.exports = {
    endpoint: endpoint
};
