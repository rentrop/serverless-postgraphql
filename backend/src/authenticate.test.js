import HttpTransport from 'lokka-transport-http';
import Lokka from 'lokka';
import { endpoint } from './endpoints';

const anonLokka = new Lokka({transport: new HttpTransport(endpoint)});
const superAdminEmail = 'superadmin@flo.ods';
const communityAdminEmail = 'admin@community.floods';
const communityEditorEmail = 'editor@community.floods';
const inactiveUserEmail = 'inactive@community.floods';
const everyPassword = 'texasfloods';
const wrongPassword = 'wrong';

function shouldWork(email, password, extra_description) {
  describe('as ' + email + ' ' + (extra_description || ''), () => {
    it('should generate a valid jwtToken', async () => {
      const response = await anonLokka.send(`
        mutation($email:String!, $password:String!) {
          authenticate(input: {email: $email, password: $password}) {
            jwtToken
          }
        }
      `,
      {
        email: email,
        password: password
      });

      expect(response.authenticate.jwtToken).not.toBeNull();
    });
  }); 
}

function shouldFail(email="", password="", extra_description) {
  describe('as ' + email + ' ' + (extra_description || ''), () => {
    it('should not generate a valid jwtToken', async () => {
      const response = await anonLokka.send(`
        mutation($email:String!, $password:String!) {
          authenticate(input: {email: $email, password: $password}) {
            jwtToken
          }
        }
      `,
      {
        email: email,
        password: password
      });

      expect(response.authenticate.jwtToken).toBeNull();
      expect(response).toMatchSnapshot();
    });
  }); 
}

describe('When authenticating', () => {
  shouldFail();
  shouldWork(superAdminEmail, everyPassword);
  shouldFail(superAdminEmail, wrongPassword);
  shouldWork(communityAdminEmail, everyPassword);
  shouldFail(communityAdminEmail, wrongPassword);
  shouldWork(communityEditorEmail, everyPassword);
  shouldFail(communityEditorEmail, wrongPassword);
  shouldFail(inactiveUserEmail, everyPassword);
});
