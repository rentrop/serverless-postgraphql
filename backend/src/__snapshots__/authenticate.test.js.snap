// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When authenticating as   should not generate a valid jwtToken 1`] = `
Object {
  "authenticate": Object {
    "jwtToken": null,
  },
}
`;

exports[`When authenticating as admin@community.floods  should not generate a valid jwtToken 1`] = `
Object {
  "authenticate": Object {
    "jwtToken": null,
  },
}
`;

exports[`When authenticating as editor@community.floods  should not generate a valid jwtToken 1`] = `
Object {
  "authenticate": Object {
    "jwtToken": null,
  },
}
`;

exports[`When authenticating as inactive@community.floods  should not generate a valid jwtToken 1`] = `
Object {
  "authenticate": Object {
    "jwtToken": null,
  },
}
`;

exports[`When authenticating as superadmin@flo.ods  should not generate a valid jwtToken 1`] = `
Object {
  "authenticate": Object {
    "jwtToken": null,
  },
}
`;
